nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ReplaceText.java

```diff
@@ -706,49 +690,43 @@ private static String normalizeReplacementString(String replacement) {
         FlowFile replace(FlowFile flowFile, ProcessSession session, ProcessContext context, String evaluateMode, Charset charset, int maxBufferSize);
 
         boolean isAllDataBufferedForEntireText();
+    }
 
-        void replaceInLine(BufferedWriter bw, String oneLine, @Nullable Matcher matcher, @Nullable Pattern searchPattern, ProcessContext context, FlowFile flowFile) throws IOException ;
+    @FunctionalInterface
+    private interface ReplaceLine {
+        void apply(BufferedWriter bw, String oneLine) throws IOException;
     }
 
 
     private class StreamReplaceCallback implements StreamCallback {
         private final Charset charset;
         private final int maxBufferSize;
-        private final ProcessContext context;
-        private final FlowFile flowFile;
-        private final ReplacementStrategyExecutor replacementStrategyExecutor;
-        private final Pattern searchPattern;
+        private final String lineByLineEvaluationMode;
+        private final ReplaceLine replaceLine;
 
-        public StreamReplaceCallback(ReplacementStrategyExecutor replacementStrategyExecutor,
-                                     Charset charset,
+        private StreamReplaceCallback(Charset charset,
                                      int maxBufferSize,
-                                     ProcessContext context,
```

Comment:
`ProcessContext` has many information. Best to avoid passing it as an argument. If a component doesn't have dependency to ProcessContext, the code look more cleaner, its purpose will be obvious, and we can write unit tests easily.

Reply:
Understood. Will follow this.